e2e_test/udf/sql_udf.slt

```diff
@@ -189,7 +202,7 @@ create function add_sub(INT, FLOAT, INT) returns float language sql as $$select
 
 # Complex types interleaving
 statement ok
-create function add_sub_types(INT, BIGINT, FLOAT, DECIMAL, REAL) returns real language sql as 'select $1 + $2 - $3 + $4 + $5';
+create function add_sub_types(INT, BIGINT, FLOAT, DECIMAL, REAL) returns double language sql as 'select $1 + $2 - $3 + $4 + $5';
```

Comment:
We will require explicit type case if the return type remains the same (i.e., `real` in the previous case), otherwise the definition can not pass the type mismatch check.